lib/actions/adapterActions.js

```diff
@@ -1097,6 +1109,42 @@ function closeSelectedAdapter(dispatch, getState) {
     });
 }
 
+export function selectedSerialPort(port) {
+    return dispatch => (
+        new Promise(() => {
+            dispatch(closeAdapter(() => {
+                getDeviceInfo(port.serialNumber)
+                    .then(deviceInfo => {
+                        dispatch(adapterGetLocalDeviceInfoAction(deviceInfo));
+                        logger.info(`Current device family is
+                            ${familyDefinitions[deviceInfo.family].split('_')[0].replace('NRF', 'nRF')}
+                            (${('0x0' + (deviceInfo.family).toString(16)).substring(-2)})`);
```

Comment:
How about something like this in deviceDefinitions.js? Then you wouldn't have to do these string operations.

```
export const familyDefinitions = {
    [nrfjprog.NRF51_FAMILY]: 'nRF51',
    [nrfjprog.NRF52_FAMILY]: 'nRF52',
    [nrfjprog.UNKNOWN_FAMILY]: 'Unknown',
};
```

Also, the programmer app (and possibly others) need the same information. Perhaps consider adding a function for getting the string representation of a device family in pc-nrfjprog-js?

Is it necessary to show the device family code (0x01 vs 0x02) to the user? I thought that was just an id used by nrfjprog to differentiate the families.

Reply:
Agreed, showing the family code is not necessary. 
nrfjprog-js could be a better place for it.

Reply:
I see, then how about the device type code?